### ISSUE TITLE
Geocode Earth - Usability and Accessibility review

### ISSUE THREAD

**Issue by RenataGegaj** [uracreative/works#47]
This is a report for a heuristic review conducted for Geocode earth website 
https://geocode.earth/
The review follows ten Usability Heuristics for User Interface Design by Nielsen Norman Group: https://www.nngroup.com/articles/ten-usability-heuristics/
Accessibility was tested using  https://accessibilityinsights.io/docs/en/web/overview
and manual keyboard testing

**Comment by RenataGegaj** [comment]
<br />
<img width="1427" alt="Screenshot 2020-05-20 at 14 31 18" src="https://user-images.githubusercontent.com/7413721/82446239-94356f80-9aa6-11ea-81a5-1b65cf966c1d.png">
<br />
"API" it's considered a technical jargon and it's not familiar to all site visitors. Also, placed there on the nav bar as a single item, doesn't provide much context about the relation with Geocode Earth.
<br /> <br /> 
Recommendation: If the "API" is a service that Geocode Earth offers, then it might make more sense to list it under services. If it's a tool targeting developers, then list it together with other developer resources, for example:
<br />
<br />
<img width="538" alt="Screenshot 2020-05-20 at 14 55 40" src="https://user-images.githubusercontent.com/7413721/82448418-fba0ee80-9aa9-11ea-841f-d774c3358e03.png">
<br />
<br />
<br />
_______________________________________________________________________________________________________
<br />
<br />

<img width="1440" alt="Screenshot 2020-05-19 at 15 28 14" src="https://user-images.githubusercontent.com/7413721/82337940-99ce7f00-99ec-11ea-98a5-188434a0fb71.png">

This doesn't give a clear example of what the product does? 
It looks like the user can click to search a location and then hit enter and waited for . He waited for some time and thought that it's not working and scrolled down This will be a problem for first-time users who are not very familiar with the field and are on the site to find out 

**Recommendation:** Having a snippet of how the API works and user can interact with it is a good idea but make sure to clarify the expectation. Replace the string "Give it a try!" with something 

_______________________________________________________________________________________________________


<img width="1440" alt="Screenshot 2020-05-19 at 15 34 04" src="https://user-images.githubusercontent.com/7413721/82337934-989d5200-99ec-11ea-96ee-44e34f5220c1.png">
<br />
"Trusted around the globe" feels odd, It looks like it's intended to be a header but it looks like plain text instead.

**Recommendation:** Change the font and make it match with "Pricing".

_______________________________________________________________________________________________________
<br />
<br />
<img width="308" alt="Screenshot 2020-05-19 at 15 41 55" src="https://user-images.githubusercontent.com/7413721/82337889-8f13ea00-99ec-11ea-8741-ca54a623d17a.png">

"Services" is core information that users would be interested to quickly access. Placing it only on the footer it might cause findability issues.

_______________________________________________________________________________________________________

<br />
<br />
<img width="1439" alt="Screenshot 2020-05-19 at 15 37 06" src="https://user-images.githubusercontent.com/7413721/82337901-91764400-99ec-11ea-9973-e12c1e814bd2.png">

**Recommendation:** "hello@your.org" might not be common sense for some. Give more information about the trial and guide the user. 

_______________________________________________________________________________________________________


<img width="1439" alt="Screenshot 2020-05-19 at 15 36 22" src="https://user-images.githubusercontent.com/7413721/82337921-95a26180-99ec-11ea-8ab4-f1c527fcdaa9.png">
Recommendation: The icons might also not be common sense for every user. Use them in addition to text, not as a replacement. Also, you might want to put more emphasis on "discounts" to make it more visible.
 
_______________________________________________________________________________________________________


<img width="1440" alt="Screenshot 2020-05-19 at 15 34 51 3" src="https://user-images.githubusercontent.com/7413721/82337924-963af800-99ec-11ea-9899-67f961f07a6a.png">
Recommendation: The descriptions "200,000 requests per month","Up to 10 requests per second" implies prior technical knowledge. Give additional information on the descriptions. Give clear expectations to users. Specify what the pricing is dedicated to since you offer more than one service. Is it the pricing for the API? Consultancy? or Other services listed on the footer?

_______________________________________________________________________________________________________


<img width="1439" alt="Screenshot 2020-05-19 at 15 36 35" src="https://user-images.githubusercontent.com/7413721/82337918-9509cb00-99ec-11ea-82fd-3b9560d532b3.png">

![photo_2020-05-19 16 14 23](https://user-images.githubusercontent.com/7413721/82337929-9804bb80-99ec-11ea-8923-10e05b4859d0.jpeg)

The Titles look like they are list items, especially on mobile. Change font/font-weight to differentiate them.

**Comment by RenataGegaj** [comment]
<img width="1438" alt="Screenshot 2020-05-20 at 16 36 13" src="https://user-images.githubusercontent.com/7413721/82463489-2f394400-9abd-11ea-977c-97ecb15075b6.png">
All interactive elements are reachable by keyboard(Tab) but their order it's not correct while testing with the extension. I tried going through them without the extension and it worked fine. It might be an error, not sure.
The "Give it a try!" example works well with the keyboard but on the voice over it doesn't have any context to it which makes it hard to understand what's happening if the user relies only on voice over for navigation.
<br/><br/>
<img width="631" alt="Screenshot 2020-05-20 at 16 27 45" src="https://user-images.githubusercontent.com/7413721/82463494-306a7100-9abd-11ea-8a2e-7ca02a59988c.png">
The form element is missing a label
<br/><br/>
Images below are elements that don't have an alt attribute or aria-label attribute does not exist or is empty
<br/><br/>
<img width="668" alt="Screenshot 2020-05-20 at 16 26 47" src="https://user-images.githubusercontent.com/7413721/82463496-31030780-9abd-11ea-992c-e43f0142b08a.png">

<img width="1302" alt="Screenshot 2020-05-20 at 16 25 43" src="https://user-images.githubusercontent.com/7413721/82463499-319b9e00-9abd-11ea-9031-0cbbe08624d0.png">
<img width="1302" alt="Screenshot 2020-05-20 at 16 24 11" src="https://user-images.githubusercontent.com/7413721/82463502-32cccb00-9abd-11ea-88e9-76ded1852cd5.png">

On the elements on the two images below, the contrast between foreground and background colors don't meet the WCAG 2 AA contrast ratio thresholds

<img width="1302" alt="Screenshot 2020-05-20 at 16 24 33" src="https://user-images.githubusercontent.com/7413721/82463501-32cccb00-9abd-11ea-98f2-49a9d5a37b34.png">

<img width="540" alt="Screenshot 2020-05-20 at 17 39 10" src="https://user-images.githubusercontent.com/7413721/82466544-d7044100-9ac0-11ea-96fe-aa30c1b1d5bb.png">

On the interactive example, form elements are missing labels. Failing success criteria
WCAG 1.3.1, WCAG 3.3.2
<img width="639" alt="Screenshot 2020-05-20 at 16 23 42" src="https://user-images.githubusercontent.com/7413721/82463505-33656180-9abd-11ea-8b02-915b2885f0fe.png">

**Comment by elioqoshi** [comment]
Small usability error from my side here as well:

On the demo component one can easily look for places which works great:
<img width="610" alt="grafik" src="https://user-images.githubusercontent.com/5436686/83965511-80667780-a8b4-11ea-9070-3c2790497a28.png">

However, when the user selects one of the suggested places, the field gets auto filled and No suggested places are displayed:

<img width="610" alt="grafik" src="https://user-images.githubusercontent.com/5436686/83965496-6f1d6b00-a8b4-11ea-9413-b3e25a013cdf.png">

This leaves a sense of brokenness to a user and should be fixed in order to show the capabilitiers of the geocoder. One way to fix that is to disable "No results were found" once a place has been selected.

**Comment by elioqoshi** [comment]
This can be closed, thanks!
